packages/php-db-import-export/src/Backend/Bigquery/ToFinalTable/IncrementalImporter.php

```diff
@@ -135,8 +136,18 @@ public function importToTable(
 
             $state->setImportedColumns($stagingTableDefinition->getColumnsNames());
         } catch (JobException|ServiceException $e) {
+            $this->bqClient->runQuery($this->bqClient->query(
+                $this->sqlBuilder->getRollbackTransaction(),
+                $session->getAsQueryOptions()
+            ));
             throw BigqueryException::covertException($e);
-        } finally {
+        } catch (Throwable $e) {
+            $this->bqClient->runQuery($this->bqClient->query(
```

Comment:
Este ma napadli, ze ci to nefailne v pripade ze nieje ziadna aktivna transakcia a ty das rollback, transakciu zacinas na riadku 83 a predtim sa robi dedup, teoreticky ked spadn tak tak to vlitne sem a nema to aktivnu transakciu. Mne nejdu nejak lokalne testy mozes to prosim vyskusat?

Reply:
good point 
![image](https://github.com/keboola/storage-backend/assets/13363655/cdcfeb3b-5265-4328-bbe1-0e0848d0f8d3)

Reply:
a s https://github.com/keboola/storage-backend/pull/118/commits/e63b7ca9a2b7e76dea4e43b9319b5642f89b649c to skonci takto , takze cajk
![image](https://github.com/keboola/storage-backend/assets/13363655/4e4f1d0d-3108-4696-8091-c7c249208afb)